scripts/setup/bootstrap.sh

```diff
@@ -56,6 +56,29 @@ _install_additional_pip_requirements() {
     unset _SETUP_PLATFORM
 }
 
+_submodules_need_updating() {
+  # Validates if a set o submodules that should always be checked out are up to date
+
+  # Pigweed will be up to date on an initial setup, however it may change over time
+  # the rest are a small subset of things that are always checked out (have no platform attachment)
+  _SUBMODULE_PATHS=(
+    "third_party/pigweed/repo"
+    "third_party/openthread/repo"
+    "third_party/editline/repo"
```

Comment:
What about zap ?

Reply:
there is no zap submodule, we get zap via cipd.